scripts/logid-generator.py

```diff
@@ -85,24 +85,46 @@ def collect_files(directory):
 
   return all_file
 
+def collect_logs_IDs(_file, log_re):
+  f = open(_file, 'r')
+  data = f.read()
+  all_zero_ID_logs = re.findall(log_re, data)
+  f.close()
+  return all_zero_ID_logs
+
 def collect_logs(_file, prefix, max_id):
   ''' Collect logs from the given file, and find the max id within this prefix by the given RE '''
+  all_zero_ID_logs = collect_logs_IDs(_file, log0_re)
+  all_non_zero_ID_logs = collect_logs_IDs(_file, log_max_re)
 
   global VERSION_SHIFT
   f = open(_file, 'r')
 
+  prev_line = ''
+  prev_num = 0
   for (i, line) in enumerate(f.readlines()):
     # Check for 0 id
-    if log0_re.search(line) != None:
-      matched_logs.append({'file': _file, 'line_num': i, 'line': line, 'new': ''})
+    for match in all_zero_ID_logs:
+      if re.sub(r'\s+', ' ', match[5]).strip() in line:
```

Comment:
introduce a name for capture group 4 and 5 for readability

(but first check if re.MULTILINE would do the trick)